stdlib/REPL/src/REPL.jl

```diff
@@ -488,10 +488,78 @@ function repl_backend_loop(backend::REPLBackend, get_module::Function)
     return nothing
 end
 
+SHOW_MAXIMUM_BYTES::Int = 20480
+
+# Limit printing during REPL display
+mutable struct LimitIO{IO_t <: IO} <: IO
+    io::IO_t
+    maxbytes::Int
+    n::Int # max bytes to write
+end
+LimitIO(io::IO, maxbytes) = LimitIO(io, maxbytes, 0)
+
+# Forward `ioproperties` onwards (for interop with IOContext)
+Base.ioproperties(io::LimitIO) = Base.ioproperties(io.io)
+
+struct LimitIOException <: Exception
+    maxbytes::Int
+end
+
+function Base.showerror(io::IO, e::LimitIOException)
+    print(io, "$LimitIOException: aborted printing after attempting to print more than $(Base.format_bytes(e.maxbytes)) within a `LimitIO`.")
+end
+
+function Base.write(io::LimitIO, v::UInt8)
+    io.n > io.maxbytes && throw(LimitIOException(io.maxbytes))
+    io.n += write(io.io, v)
+end
+
+# Semantically, we only need to override `Base.write`, but we also
+# override `unsafe_write` for performance.
+function Base.unsafe_write(io::Base.IOContext{<:LimitIO}, p::Ptr{UInt8}, nb::UInt)
```

Comment:
Why is this method for `Base.IOContext{<:LimitIO}` instead of just for `LimitIO`?

Reply:
because we need to be able to pass IO properties since the REPL uses them, so then we have the option of either passing them on the outside (`Base.IOContext{<:LimitIO}`) or the inside (`LimitIO{<:Base.IOContext}`). I chose the outside (see line 551 below) since it was easier to make everything work correctly that way

Reply:
That makes sense but `unsafe_write` on an `IOContext` should automatically fallback to calling `unsafe_write` on the underlying `IO` https://github.com/JuliaLang/julia/blob/bec4702fa1eb76942e75285ac9ca8aa020b39c31/base/io.jl#L452

Reply:
true, maybe this method should be for `unsafe_write(io::LimitIO, p::Ptr{UInt8}, nb::UInt)` then as you say and rely on that fallback. I think in that case we won't be passing along the IO properties anymore (since the fallback drops them) but that's probably fine at this stage

Reply:
this change seemed fine locally, so I pushed it up